designsystem/src/main/java/cmm/apps/designsystem/EsmorgaButton.kt

```diff
@@ -1,23 +1,27 @@
 package cmm.apps.designsystem
 
-import androidx.compose.foundation.background
 import androidx.compose.foundation.layout.fillMaxWidth
 import androidx.compose.foundation.shape.RoundedCornerShape
 import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
 import androidx.compose.material3.MaterialTheme
 import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
 import androidx.compose.ui.unit.dp
 
 @Composable
-fun EsmorgaButton(text: String, primary: Boolean = true, onClick: () -> Unit) {
+fun EsmorgaButton(text: String, modifier: Modifier = Modifier, primary: Boolean = true, onClick: () -> Unit) {
     Button(
-        modifier = Modifier
-            .clip(RoundedCornerShape(8.dp))
-            .background(if (primary) MaterialTheme.colorScheme.primary else MaterialTheme.colorScheme.secondary)
+        shape = RoundedCornerShape(5.dp),
+        modifier = modifier
             .fillMaxWidth(),
+        colors = ButtonDefaults.buttonColors().copy(
+            contentColor = if (primary) Color.White else Color.Black,
```

Comment:
you should use `MaterialTheme.colorScheme.onPrimary` instead of a hardcoded color.